src/Symfony/Component/Messenger/Stamp/DelayStamp.php

```diff
@@ -33,16 +43,42 @@ public function getDelay(): int
 
     public static function delayForSeconds(int $seconds): self
     {
-        return new self($seconds * 1000);
+        return self::delayFor($seconds, self::PERIOD_SECONDS);
     }
 
     public static function delayForMinutes(int $minutes): self
     {
-        return self::delayForSeconds($minutes * 60);
+        return self::delayFor($minutes, self::PERIOD_MINUTES);
     }
 
     public static function delayForHours(int $hours): self
     {
-        return self::delayForMinutes($hours * 60);
+        return self::delayFor($hours, self::PERIOD_HOURS);
+    }
+
+    public static function delayUntil(\DateTimeInterface $executeAfter): self
+    {
+        $now = (new \DateTimeImmutable())->setTimezone($executeAfter->getTimezone())->format('U');
+        $diff = abs($executeAfter->format('U') - $now) * 1000;
+
+        return new self($diff);
+    }
+
+    /**
+     * @param string $period A string representing a unit symbol valid for relative formats of DateTime objects
+     *
+     * @see https://www.php.net/manual/en/datetime.formats.relative.php#datetime.formats.relative
+     */
+    public static function delayFor(int $units, string $period)
```

Comment:
Please remove

Reply:
I'm not understanding: why?

Reply:
We dont need to reinvent a way to define time intervals. We got the `DateInterval`, Either using that or a DateTime is fine.

Reply:
So, should have I to add `delayForDays()` method?

Reply:
It is simply a shortcut, like the ones already exists in the class.